tests/chainer_tests/functions_tests/math_tests/test_sum.py

```diff
@@ -11,18 +11,29 @@
 from chainer.testing import condition
 
 
+@testing.parameterize(
+    {'dtype': numpy.float16},
+    {'dtype': numpy.float32},
+    {'dtype': numpy.float64},
+)
```

Comment:
OK, i'll fix to use `product`.
Your suggestion is Ok, but I think that is another issue. I'll send another PR about shapes.